MobSF/views/api/rest_api.py

```diff
@@ -181,32 +179,22 @@ def api_json_report(request):
     return response
 
 
-@request_method(['GET'])
-@csrf_exempt
-def api_viewsource_android(request):
-    """
-    viewsource for android file
-    """
-    viewsource_form = ViewSourceAndroidForm(request.GET)
-    if not viewsource_form.is_valid():
-        return JsonResponse(FormUtil.errors_message(viewsource_form), status=BAD_REQUEST)
-
-    view_source = ViewSourceAndroid(request)
-    return view_source.api()
-
-
-
-@request_method(['GET'])
+@request_method(['POST'])
 @csrf_exempt
-def api_viewsource_ios(request):
+def api_view_source(request):
     """
-    viewsource for ios file
+    View Source for android & ios source file
     """
-    viewsource_form = ViewSourceIosForm(request.GET)
-    if not viewsource_form.is_valid():
-        return JsonResponse(FormUtil.errors_message(viewsource_form), status=BAD_REQUEST)
-
-    view_source = ViewSourceIos(request)
-    return view_source.api()
-    
-
+    params = ['file', 'type', 'hash']
+    if set(request.POST) >= set(params):

```

Comment:
It seems that there is no need to check the parameters here.

Reply:
You are right. But because we have form checks in the other function. But since other functions use the same pattern and they don't have form validation. I thought to keep it as this should not confuse anyone in future.

I will remove that probably later when we have form checks for every API. We will do that as a part of code QA.